src/Tests/Tests/XPack/CrossClusterReplication/CrossClusterReplicationAutoFollowTests .cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Elastic.Xunit.XunitPlumbing;
+using FluentAssertions;
+using Nest;
+using Tests.Core.ManagedElasticsearch.Clusters;
+using Tests.Core.ManagedElasticsearch.NodeSeeders;
+using Tests.Domain;
+using Tests.Framework;
+using Tests.Framework.EndpointTests.TestState;
+using Tests.Framework.Integration;
+
+namespace Tests.XPack.CrossClusterReplication
+{
+	[SkipVersion("<6.5.0", "")]
+	public class CrossClusterReplicationAutoFollowTests : CoordinatedIntegrationTestBase<WritableCluster>
```

Comment:
Similar to other comment:

As I understand it, each step is dependent on the "setup" of the previous one, and is asserted in its own integration test method in the class.

If my understanding is correct, maybe we could do something similar here as to what exists for ML job API integration tests - a base class with setup steps that can be called in IntegrationSetup method for each deriving integration test, and a derived test class for each separate API?

Reply:
Aye, The ML approach is valid since most operate on a job so their `IntegrationSetup` is relatively straightforward and inheritable.

The setup in ML run constantly though for each `ApiTest` test class.

CCR is more a chain of API's to call, where the same approach would leave a lot of unfinished setup chains e.g `CoodinatedIntegrationTest` allows us to express this dependency chain:

`A1 => A2 => A3 => A4 = > A5 => A6`

Which will be only executed once for each permutation of `[async, sync]` and `[fluent, ois]`

if we take the object graph approach to test each `AN` in isolation

We end up with:

```
A1
A1 => A2
A1 => A2 => A3 
A1 => A2 => A3 => A4 
A1 => A2 => A3 => A4 = > A5 
A1 => A2 => A3 => A4 = > A5 => A6
```

Foreach permutation of `[async, sync]` and `[fluent, ois]`.